dial_test.go

```diff
@@ -558,14 +519,13 @@ func newSilentListener(t *testing.T) ([]ma.Multiaddr, net.Listener) {
 }
 
 func TestDialSimultaneousJoin(t *testing.T) {
-	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
why change this to background? A fault could leave dangling dials.
But then again it maybe doesnt matter.

Reply:
It can't. This test blocks until all 3 go routine have returned:
```go
// raise any errors from the previous goroutines
for i := 0; i < 3; i++ {
	require.NoError(t, <-errs)
}
```